### PR TITLE
Nutrition fact CRUD (w/ products)

### PR DESCRIPTION
**Pull request by tien4112004**
## Changes

* Add NutritionFact CRUD in product related route. 
* Add checkout route for order.

### CODE REVIEW

File: BistroQ.Services/Mappings/MappingProfile.cs [hunk 1]
```diff
@@ -17,13 +19,31 @@ public MappingProfile()
     {
         CreateMap<Product, ProductDto>().ReverseMap();
         CreateMap<CreateProductRequestDto, Product>()
```
Comment (Tondeptrai23) [inline BistroQ.Services/Mappings/MappingProfile.cs:21]: Why was `AfterMap` used here? `AutoMapper` already supports nested mapping. 
Could it be that `ConstructUsing` disables this functionality?

Comment (tien4112004) [reply BistroQ.Services/Mappings/MappingProfile.cs:21]: To fix the error with request form-data, I have to map every nested items to top-level. Nested mapping does not worked.

Comment (tien4112004) [reply BistroQ.Services/Mappings/MappingProfile.cs:21]: > To fix the error with request form-data, I have to map every nested items to top-level. Nested mapping does not worked.

```csharp
public class CreateProductRequestDto
{
    public string Name { get; set; }
    public decimal Price { get; set; }
    public string Unit { get; set; }
    public decimal DiscountPrice { get; set; }
    public int? CategoryId { get; set; }
    public double Calories { get; set; }
    public double Fat { get; set; }
    public double Fiber { get; set; }
    public double Protein { get; set; }
}
```

`NutritionFact` related fields does not belongs to any object, as you can see.

Comment (Tondeptrai23) [reply BistroQ.Services/Mappings/MappingProfile.cs:22]: Ok

**Review by Tondeptrai23 — APPROVED** [review]

**Pull request merged**